src/CrossmintEmbed.ts

```diff
@@ -88,15 +117,16 @@ export default class CrossmintEmbed {
             }
 
             window.removeEventListener("message", handleMessage);
-            resolve(_accounts);
+            resolve(loginData);
         });
     }
 
-    async signMessage(message: Uint8Array): Promise<Uint8Array | undefined | null> {
+    signMessage<T>(message: Uint8Array, walletId?: string, deviceId?: string): Promise<T>
+    async signMessage(message: Uint8Array, walletId?: string, deviceId?: string) {
```

Comment:
would this work instead?
```suggestion
    async signMessage<T>(message: Uint8Array, walletId?: string, deviceId?: string): Promise<T> {
```

Reply:
related to above comment